lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2344,51 +2346,44 @@ mkApiTransactionFromInfo ti wrk wid deposit info metadataSchema = do
         Expired  -> #pendingSince
 
 postTransactionFeeOld
-    :: forall s k n
+    :: forall s k ktype (n :: NetworkDiscriminant)
      . ( Typeable n
        , Typeable k
        , Typeable s
+       , AddressBookIso s
        , BoundedAddressLength k
        )
-    => ApiLayer s k 'CredFromKeyK
+    => ApiLayer s k ktype
     -> ApiT WalletId
     -> PostTransactionFeeOldData n
     -> Handler ApiFee
-postTransactionFeeOld ctx@ApiLayer{..} (ApiT wid) body =
-    withWorkerCtx ctx wid liftE liftE $ \wrk -> do
-        let db = wrk ^. dbLayer
-        era <- liftIO $ NW.currentNodeEra netLayer
+postTransactionFeeOld ctx@ApiLayer{..} (ApiT walletId) body = do
+    protocolParameters <- liftIO $ currentProtocolParameters netLayer
+    era <- liftIO $ NW.currentNodeEra netLayer
+    AnyRecentEra (recentEra :: WriteTx.RecentEra era) <- guardIsRecentEra era
+    withWorkerCtx ctx walletId liftE liftE $ \workerCtx -> do
+        let db = workerCtx ^. dbLayer
         wdrl <- case body ^. #withdrawal of
             Nothing -> pure NoWithdrawal
             Just apiWdrl ->
-                shelleyOnlyMkWithdrawal @s @k @n @'CredFromKeyK
-                    netLayer txLayer db wid era apiWdrl
-        let txCtx = defaultTransactionCtx
+                shelleyOnlyMkWithdrawal @s @k @n
+                    netLayer txLayer db walletId era apiWdrl
+        let outs = addressAmountToTxOut <$> body ^. #payments
+        minCoins <- liftIO $ W.calcMinimumCoinValues @_ @k @ktype
+            workerCtx era (F.toList outs)
+        feePercentiles <- liftHandler $ W.transactionFee @s @k @ktype @n
+            db
+            protocolParameters
+            txLayer
+            (timeInterpreter netLayer)
+            (AnyRecentEra recentEra)
+            (dummyChangeAddressGen @k)
+            walletId
+            defaultTransactionCtx
                 { txWithdrawal = wdrl
                 , txMetadata = body
                     ^? #metadata . traverse . #txMetadataWithSchema_metadata
                 }
-        (utxoAvailable, wallet, pendingTxs) <-
-            liftHandler $ W.readWalletUTxOIndex @_ @s @k wrk wid
-        let outs = addressAmountToTxOut <$> body ^. #payments
-        pp <- liftIO $ NW.currentProtocolParameters netLayer
-        let selectAssetsParams = W.SelectAssetsParams
-                { outputs = F.toList outs
```

Comment:
Pretty sure we need to include the `outs` in the `PreSelection` inside `transactionFee`